### PR TITLE
ncnews 14

### PR DESCRIPTION
**Pull request by timwtuck**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -72,6 +72,41 @@ describe('# GET REQUESTS', () => {
...
+                });
+        });
+        test('400: invalid query', () => {
+            return request(app)
+                .get('/api/articles?sort_by=not_valid_column')
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Invalid Query Item");
+                });
+        });
```
Comment (jadelandeg) [inline __tests__/app.test.js:109]: add another test for an invalid order by

also a 404 for non existent topic

File: models/articles.models.js [hunk 2]
```diff
@@ -18,14 +18,17 @@ exports.selectArticle = async (id) => {
...
 }
 
-exports.selectAllArticles = async () => {
+exports.selectAllArticles = async (sortBy = 'created_at', order = 'desc', topic = '%%') => {
+
+    await this._validateInput({sortBy, order});
     
     const query = `SELECT articles.*, COUNT(comment_id)::INTEGER AS comment_count FROM articles
                     LEFT JOIN comments ON comments.article_id = articles.article_id
+                    WHERE topic ILIKE $1
```
Comment (jadelandeg) [inline models/articles.models.js:27]: this where should be conditional

**Review by jadelandeg — CHANGES_REQUESTED** [review]
couple of changes! :)

File: models/articles.models.js (hunk 2 above)
Comment (jadelandeg) [inline models/articles.models.js:28]: oh my bad, just realised what you have done so ignore previous comment. All good

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**